src/parser.rs

```diff
@@ -125,6 +141,32 @@ pub fn inline<T>(result: Result<T, ParserError>) -> Result<T, (ParserError, Toke
 	result.map_err(|error| (error, Token::Terminator))
 }
 
+pub fn animation(lexer: &mut Lexer, check_with: bool) -> Result<Option<AnimationDeclaration>, (ParserError, Token)> {
```

Comment:
Do we need to have the `check_with` parameter inside here? It's only used in the `spawn` command.

Reply:
How else would you suggest it be done? The other way would require checking for a `with` identifier for every command.
I initially had an `animation_declaration` function which would *only* parse the `glide[]` part which the `animation` function would call after checking for a `with` identifier. Would you like me to switch back to that? It would look like:
```rust
pub fn animation(lexer: &mut Lexer) -> Result<Option<AnimationDeclaration>, (ParserError, Token)> {
  // Checks for with
  animation_declaration(lexer)
}

pub fn animation_declaration(lexer: &mut Lexer) -> Result<Option<AnimationDeclaration>, (ParserError, Token)> {
  // Code
}
```

Reply:
Added a `peek` method as recommended by you on Discord.